plotly/tests/test_optional/test_figure_factory/test_figure_factory.py

```diff
@@ -2062,45 +2091,30 @@ def test_valid_facet_grid_fig(self):
         )
 
         exp_facet_grid = {
-            'data': [{'marker': {'color': 'rgb(31, 119, 180)',
-                'line': {'color': 'darkgrey', 'width': 1},
-                'size': 8},
+            'data': [
+              {'marker': {'color': 'rgb(31,119,180)',
+               'line': {'color': 'darkgrey', 'width': 1},
+               'size': 8},
                'mode': 'markers',
                'opacity': 0.6,
                'type': 'scatter',
-               'x': [1.8,
-                    1.8,
-                    2.0,
-                    2.0,
-                    1.8,
-                    1.8,
-                    2.0],
-               'xaxis': 'x1',
-               'y': [18,
-                    21,
-                    20,
-                    21,
-                    18,
-                    16,
-                    20],
-               'yaxis': 'y1'},
-              {'marker': {'color': 'rgb(31, 119, 180)',
+               'x': [1.8, 1.8, 2.0, 2.0, 1.8, 1.8, 2.0],
+               'xaxis': 'x',
+               'y': [18, 18, 20, 21, 18, 16, 20],
+               'yaxis': 'y'},
+              {'marker': {'color': 'rgb(31,119,180)',
                 'line': {'color': 'darkgrey', 'width': 1},
                 'size': 8},
                'mode': 'markers',
                'opacity': 0.6,
                'type': 'scatter',
-               'x':     [2.8,
-                   2.8,
-                   3.1],
+               'x': [2.8, 2.8, 3.1],
                'xaxis': 'x2',
-               'y':    [16,
-                   18,
-                   18],
-               'yaxis': 'y1'}],
+               'y': [16, 18, 18],
+               'yaxis': 'y'}],
              'layout': {'annotations': [{'font': {'color': '#0f0f0f', 'size': 13},
                 'showarrow': False,
-                'text': 4,
+                'text': '4',
```

Comment:
🚨 Breaking Change - `go.Annotation.text` has to be a string now

Although maybe we should update the plotschema for this one as well, since technically I think it can be a number. We made a proposal for updating the schema here: https://github.com/plotly/plotly.js/issues/2519

Reply:
https://github.com/plotly/plotly.js/issues/2523

Reply:
@chriddyp @Kully I just pushed ccbd940df2cd576d5de1e7ef650bc602aceb67bc which supports the `strict` option discussed in plotly/plotlyjs#2523. Now numbers should be accepted as strings for most string properties.  So we should be able to revert the number->str testing changes now.

Reply:
Oh I see, this change is in the 'expected' result.  With my latest change numbers are accepted by non-strict string properties (like annotation.text), but they are converted to strings internally.